crates/cheatable-starknet/src/cheatcodes/mock_call.rs

```diff
@@ -0,0 +1,35 @@
+use crate::{cheatcodes::EnhancedHintError, CheatedState};
+use starknet_api::core::{ContractAddress, EntryPointSelector};
+use starknet_api::hash::StarkFelt;
+use std::collections::HashMap;
+
+impl CheatedState {
+    pub fn start_mock_call(
+        &mut self,
+        contract_address: ContractAddress,
+        function_name: EntryPointSelector,
+        ret_data: Vec<StarkFelt>,
+    ) -> Result<(), EnhancedHintError> {
+        self.mocked_functions.entry(contract_address).or_insert_with(|| HashMap::new());
+
+        let contract_mocked_fns = self.mocked_functions.get_mut(&contract_address).unwrap();
```

Comment:
Would use `contract_mocked_functions`, I had to think for a while what `fns` mean